File: src/main/scala/org/apache/datacommons/prepbuddy/clusterers/Cluster.scala
package org.apache.datacommons.prepbuddy.clusterers

import scala.collection.mutable.ListBuffer

/**
  * Cluster contains groups of values by their specified key
  */
class Cluster(key: String) extends Serializable {

    private val tuples: ListBuffer[(String, Int)] = ListBuffer.empty

    def contain(tuple: (String, Int)): Boolean = tuples.contains(tuple)

    def size: Int = tuples.size

    def isOfKey(key: String): Boolean = key.equals(this.key)

    def containsValue(value: String): Boolean = tuples.exists(_._1 == value)

    def add(tuple: (String, Int)): Unit = tuples += tuple

}

File: python/pyprepbuddy/cluster/cluster.py
class Cluster(object):
    """
    Cluster contains groups of values by their specified key
    """
    def __init__(self, cluster):
        self.__cluster = cluster

    def __contains__(self, item):
        return self.__cluster.containsValue(item)

    def size(self):
        return self.__cluster.size()

    def get_cluster(self):
        return self.__cluster
